### DISCUSSION TITLE
Props types, like 'Number' and 'Array', are removed by genRuntimePropDefinition

### DISCUSSION THREAD

**Discussion by ChloeFancy** [vue-macros/vue-macros#779] [General]
https://github.com/vue-macros/vue-macros/blob/7a0864635786ffd59858b0892e98606ececa2460/packages/api/src/vue/utils.ts#L144-L176

confused about why `genRuntimePropDefinition` intend to filter types causing types like 'Number' and 'Array' get removed, in L157-L160

**Comment by sxzz** [comment]
Only `Boolean`, `Function`, and `String` are meaningful in production mode. They are also removed in the official version.
https://play.vuejs.org/#__PROD__eNp9kT9PwzAQxb+K5amVqjDAFKVBgDrAABUwekmTS3BxfMZ/QkTJd+ecqClD1c333jvd784HfmdM0gXgKc9caaXxzIEPhqlCN2vBvRM8F7qCWmrYWjQuOwjNWI14mzLnrdRNrHeFpXqHqKDQk/BDwmLJ1jnrUFZR+wowN7HfU3rIF0uhs6sJgMbxFQ0uUdeySfYONdGNUwUvsTVSgX0xXqImuJSNTvQKpfD7adS8DbA66uUHlJ9n9L3royb41oID24Hgs+cL24Cf7M3bM/T0ns0Wq6AofcF8BYcqRMYpdh90Rdj/ciPtY2vQejrHu9v0HrQ7LhVBY3IY84LTFz1cWP2Ee53cjH10VT78AcmZnTg=

**Comment by CyanSalt** [comment]
This may be a meaningful message: for third-party library developers (such as UI component libraries), non-production environment flags **should always be passed**. Otherwise, applications using these UI components will receive development warnings when passing props such as `number` types to components.

Perhaps it would be better for packages such as `@vue-macro/better-define` to actively identify Vite Library Mode and handle these cases by default.

**Reply by sxzz** [reply]
To run in a non-production environment, set `isProduction: false`. When `isProduction` is false, lines 157-160 won't execute, leaving all types unchanged.

By default, `isProduction` depends on `process.env.NODE_ENV`, but you can override it by setting `isProduction: false`.

```ts
VueBetterDefine({ isProduction: false })
```

**Reply by CyanSalt** [reply]
https://github.com/vue-macros/vue-macros/blob/3ca89953ce484c85ec15a7d1aaf1b4447b83169a/packages/better-define/src/index.ts#L73-L75

@sxzz This doesn't work properly in Vite Library Mode, because `@vue-macros/better-define` always uses vite's flag as the value of `isProduction` unless I build my UI library via `NODE_ENV=development vite build`.

A more reasonable priority seems to be: userOptions > vite config > default (by `NODE_ENV`).
